### ISSUE TITLE
implementation of multi-dimensional data types

### ISSUE THREAD

**Issue by dev-gb** [modelica/DCPLib#16]
I need a bit of help with the implementation of multidimensional arrays in the dcpx description file and in the slave example. 

I tried to create these variables below.
```
robot[...].speed
          .charge
          ...
```       
The variables should be inputs to my dcp slave.
Does anyone have similar example code? 
 
Many thanks in advance.

**Comment by martinkrammer** [comment]
You need to specify Input/Dimensions/... in the DCPX file

**Comment by dev-gb** [comment]
That's what I thought... 

A implementation example in the C++ library or a wiki entry would be very helpfull.
Especially reading and writing dimension variables is not clear to me.

The dcpx file can be created herewith.
``` c++		
caus_RobotListDimension->Uint16->min= std::make_shared<uint16_t>(0);
caus_RobotListDimension->Uint16->max= std::make_shared<uint16_t>(122);
caus_RobotListDimension->dimensions.push_back(make_Dimension(DimensionType::LINKED_VR, charge_vr));
caus_RobotListDimension->dimensions.push_back(make_Dimension(DimensionType::LINKED_VR, velocity_vr));
Variable_t RobotListDimensionPar= make_Variable_input("Robot[...]", RobotListDimension_vr, caus_RobotListDimension);
```

**Comment by hannesha** [comment]
I just discovered that multidimensional variables as inputs are broken at the moment.
The received payload size is only compared with the datatype size instead of the payload size from the multidimvalue.

I'm already working on a fix.

**Comment by clemensschiffer** [comment]
I have fixed some bugs for multidimensional variables and implemented an example to show their usage in #52
